core/src/main/java/org/elasticsearch/action/termvectors/TransportMultiTermVectorsAction.java

```diff
@@ -80,7 +79,7 @@ protected void doExecute(final MultiTermVectorsRequest request, final ActionList
                 continue;
             }
             ShardId shardId = clusterService.operationRouting().getShards(clusterState, concreteSingleIndex,
```

Comment:
get shards here is an overkill. Let's add a operationRouting().resolveShard() method that returns a shardId.